pkg/resources/custom_diffs.go

```diff
@@ -195,3 +195,32 @@ func RecreateWhenUserTypeChangedExternally(userType sdk.UserType) schema.Customi
 		return nil
 	}
 }
+
+func RecreateWhenSecretTypeChangedExternally(secretType string, oauthType *sdk.OauthSecretType) schema.CustomizeDiffFunc {
+	return func(_ context.Context, diff *schema.ResourceDiff, _ interface{}) error {
+		if n := diff.Get("secret_type"); n != nil {
+			logging.DebugLogger.Printf("[DEBUG] new external value for secret type %s\n", n.(string))
+			if secretType != strings.ToUpper(n.(string)) {
```

Comment:
Instead of comparing strings here, we can introduce a function converting from a string to a secret type - just like in other resources.